### PR TITLE
updating KMC to version 3.1.1rc1

### PR DESCRIPTION
**Pull request by notestaff**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [X] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [X] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/kmc/3.1.1rc1/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  patches:
+    - 0001-fixes-for-conda.patch
+
+build:
+  number: 0
+  # this recipe builds kmc without the asmlib speedup
+
+requirements:
+  build:
+    # C++14 features are required
```
Comment (bgruening) [inline recipes/kmc/3.1.1rc1/meta.yaml]: Are you sure about this comment? You are not using the new compilers as it seems.

Comment (notestaff) [reply recipes/kmc/3.1.1rc1/meta.yaml]: @bgruening the makefile does specify C++14: https://github.com/bioconda/bioconda-recipes/pull/9952/files#diff-10ea6eb45d548abe3d44bba556156971R32

**Comment by bgruening** [comment]
I see, it is blacklisted. Can you please remove kmc from the blacklist in this branch. Thanks.

**Comment by notestaff** [comment]
@bgruening so, is it possible to build with C++14?  if so, how?

**Comment by bgruening** [comment]
Yes, include this file https://github.com/bioconda/bioconda-recipes/blob/master/recipes/stacks/2.1/conda_build_config.yaml into your folder. It will overwrite the default settings.

**Comment by bgruening** [comment]
Both version will be build now. The older one fails because of the missing new compiler, the newer one fails because of a missing patch as it seems.
If you like you can remove the older recipe.

**Comment by bgruening** [comment]
Thanks!!!

**Pull request merged**